src/boxy/textures.nim

```diff
@@ -136,9 +128,58 @@ proc updateSubImage*(texture: Texture, x, y: int, image: Image) =
     y = y div 2
     inc level
 
+proc clearSubImage*(texture: Texture, x, y, width, height: int, level: int = 0) =
```

Comment:
Could save a lot of work by not needing these, or keeping it only for debug

Reply:
We need to clear the margins if images are smaller.... there are probably faster way to do that as you said before margin clear rects.

Reply:
yeah i would just draw margins before adding images, and drop the clear entirely

Reply:
(and why the background is usually drawn last in 3D; draw all the geo to a depth-only attachment, then draw it in shader-optimal order where depth-test saves on lots of overdraw, and then draw the background only on pixels where the depth value == infinity)